spec/hardware/system.fmf

```diff
@@ -1,12 +1,17 @@
 summary:
     Select machine with given system attributes
-description:
-    Use the ``vendor`` key to choose system by specific vendor,
-    ``model`` to provide system's model name, ``numa-nodes``
-    to select systems with a given number of NUMA nodes.
 example:
   - |
     system:
-        vendor: HPE
-        model: ProLiant DL385 Gen10
+        # System vendor by ID...
+        vendor: 0x03E8
```

Comment:
@happz so json schema obviously cannot recognize hexadecimal numbers, so is there some code which would then prevent putting some arbitrary no number string?

Reply:
Well, this just an example, it can be replaced by `12345`.

In general, when the value of the string cannot be converted to an integer and exception is raised reporting the specification is invalid.

This may change, there might be exotic IDs, e.g. `12:34:56` might be perfectly valid ID on some strange bus, and the corresponding code handling the key would take care of raising "specification invalid" exceptions.